components/size_aware_image.jsx

```diff
@@ -170,6 +196,79 @@ export default class SizeAwareImage extends React.PureComponent {
             />
         );
 
+        // copyLink, download are two buttons overlayed on image preview
+        // copyLinkTooltip, downloadTooltip are tooltips for the buttons respectively.
+        // if linkCopiedRecently is true, defaultMessage would be 'Copy Link', else 'Copied!'
+
+        const copyLinkTooltip = (
+            <Tooltip
+                id='copy-link-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.copy_link_tooltip'
+                    defaultMessage={this.state.linkCopiedRecently ? 'Copied' : 'Copy link'}
+                />
+            </Tooltip>
+        );
+        const copyLink = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={copyLinkTooltip}
+                rootClose={true}
+            >
+                <button
+                    className={classNames('style--none', 'size-aware-image__copy_link', {
+                        'size-aware-image__copy_link--recently_copied': this.state.linkCopiedRecently,
+                    })}
+                    aria-label={`${ariaLabelImage}${localizeMessage('copy url')}`}
+                    onClick={this.copyLinkToAsset}
+                >
+                    {this.state.linkCopiedRecently ? (
+                        <i className='icon icon-check style--none'/>
+                    ) : (
+                        <i className='icon icon-link-variant style--none'/>
+                    )}
+                </button>
+            </OverlayTrigger>
+        );
+
+        const downloadTooltip = (
+            <Tooltip
+                id='download-preview-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.download_tooltip'
+                    defaultMessage='Download'
+                />
+            </Tooltip>
+        );
+
+        const download = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={downloadTooltip}
+                rootClose={true}
+            >
+                <a
+                    href={this.isInternalImage() ? fileURL : src}
+                    className='style--none size-aware-image__download'
+                    target='_blank'
+                    rel='noopener noreferrer'
+                    download={ariaLabelImage}
```

Comment:
I missed this earlier, but passing a value for this changes the name of the file being downloaded which I don't think we want. I think just passing `download={true}` here will cause the original name of the file to be preserved, but we should confirm that that works for both internal and external images